### PR TITLE
chore: update java-operator-sdk to v3, quarkus-operator-sdk to v4

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Review by rinaldodev — APPROVED** [review]

File: cos-fleetshard-api/runtime/pom.xml [hunk 1]
```diff
@@ -31,7 +31,7 @@
         <plugins>
             <plugin>
                 <groupId>io.quarkus</groupId>
-                <artifactId>quarkus-bootstrap-maven-plugin</artifactId><executions>
+                <artifactId>quarkus-extension-maven-plugin</artifactId><executions>
```
Comment (rinaldodev) [inline cos-fleetshard-api/runtime/pom.xml:34]: Is this related?

Comment (lburgazzoli) [reply cos-fleetshard-api/runtime/pom.xml:34]: yes, the bootstrap plugin has been deprecated in recent quarkus version in favor of the extension plugin
